script/SingletonDeployer_v1.s.sol

```diff
@@ -0,0 +1,275 @@
+import {InverterReverter_v1} from
+    "src/external/reverter/InverterReverter_v1.sol";
+import {FeeManager_v1} from "@ex/fees/FeeManager_v1.sol";
+import {TransactionForwarder_v1} from
+    "@ex/forwarder/TransactionForwarder_v1.sol";
+
+// Beacon
+import {
+    InverterBeacon_v1,
+    IInverterBeacon_v1
+} from "src/proxies/InverterBeacon_v1.sol";
+
+// Modules
+// =============================================================================
+import {Orchestrator_v1} from "src/orchestrator/Orchestrator_v1.sol";
+import {IModule_v1} from "src/modules/base/IModule_v1.sol";
+
+// Funding Managers
+import {FM_Rebasing_v1} from "@fm/rebasing/FM_Rebasing_v1.sol";
+import {FM_BC_Bancor_Redeeming_VirtualSupply_v1} from
+    "@fm/bondingCurve/FM_BC_Bancor_Redeeming_VirtualSupply_v1.sol";
+import {FM_BC_Restricted_Bancor_Redeeming_VirtualSupply_v1} from
+    "@fm/bondingCurve/FM_BC_Restricted_Bancor_Redeeming_VirtualSupply_v1.sol";
+import {BancorFormula} from "@fm/bondingCurve/formulas/BancorFormula.sol";
+
+// Authorization
+import {AUT_Roles_v1} from "@aut/role/AUT_Roles_v1.sol";
+import {AUT_TokenGated_Roles_v1} from "@aut/role/AUT_TokenGated_Roles_v1.sol";
+import {AUT_EXT_VotingRoles_v1} from
+    "src/modules/authorizer/extensions/AUT_EXT_VotingRoles_v1.sol";
+
+// Payment Processors
+import {PP_Simple_v1} from "src/modules/paymentProcessor/PP_Simple_v1.sol";
+import {PP_Streaming_v1} from "src/modules/paymentProcessor/PP_Streaming_v1.sol";
+
+// Logic Modules
+import {LM_PC_Bounties_v1} from "@lm/LM_PC_Bounties_v1.sol";
+import {LM_PC_RecurringPayments_v1} from "@lm/LM_PC_RecurringPayments_v1.sol";
+import {LM_PC_Staking_v1} from "@lm/LM_PC_Staking_v1.sol";
+import {LM_PC_KPIRewarder_v1} from "@lm/LM_PC_KPIRewarder_v1.sol";
+import {LM_PC_PaymentRouter_v1} from "@lm/LM_PC_PaymentRouter_v1.sol";
+
+// Factories
+import {ModuleFactory_v1} from "src/factories/ModuleFactory_v1.sol";
+import {OrchestratorFactory_v1} from "src/factories/OrchestratorFactory_v1.sol";
+
+//--------------------------------------------------------------------------
+// General Singelton Deployer Information
+//--------------------------------------------------------------------------
+// This file acts as a general deployer for all singeltons in the inverter protocol.
+// It uses a deterministic factory to deploy all contracts needed for the inverter protocol to work.
+//--------------------------------------------------------------------------
+
+contract SingletonDeployer_v1 is ProtocolConstants_v1 {
+    using Strings for string;
+
+    IDeterministicFactory_v1 public factory =
+        IDeterministicFactory_v1(deterministicFactory);
+
+    bytes32 factorySalt = keccak256(abi.encodePacked("inverter-deployment"));
+
+    //External contracts
+
+    address public ext_InverterReverter_v1;
+    address public ext_Governor_v1;
+    address public ext_TransactionForwarder_v1;
+    address public ext_FeeManager_v1;
+
+    //@note Add IssuanceToken here?
```

Comment:
I don't think so. Each project should create its own IssuanceToken. We could create a factory contract for it, but in my opinion this is overkill for now. Fabi's PIM factory should be enough for now.